### PR TITLE
consider security disabled case in permission check

### PR DESCRIPTION
**Pull request by joshuali925**
### Description
see #9 
rebased #9 to agent framework branch

### Issues Resolved
_List any issues this PR will resolve, e.g. Closes [...]._ 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: public/plugin.tsx [hunk 1]
```diff
@@ -46,23 +51,28 @@ export class AssistantPlugin
...
       return async () => {
-        if (enabled === undefined) {
-          const account = await getAccount();
-          enabled = account.data.roles.some((role) =>
-            ['all_access', 'assistant_user'].includes(role)
-          );
+        if (setupDeps.securityDashboards === undefined)
+          return { data: { roles: ['all_access'], user_name: 'dashboards_user' } };
+        if (account === undefined) {
+          account = await core.http
```
Comment (wanglam) [inline public/plugin.tsx:64]: Seems the `account` was not be modified in other place, shall we change to `const account`?

File: public/plugin.tsx [hunk 2]
```diff
@@ -46,23 +51,28 @@ export class AssistantPlugin
...
+      let account: UserAccountResponse;
       return async () => {
-        if (enabled === undefined) {
-          const account = await getAccount();
-          enabled = account.data.roles.some((role) =>
-            ['all_access', 'assistant_user'].includes(role)
-          );
+        if (setupDeps.securityDashboards === undefined)
+          return { data: { roles: ['all_access'], user_name: 'dashboards_user' } };
+        if (account === undefined) {
```
Comment (wanglam) [inline public/plugin.tsx:63]: The `account` is always be undefined here, shall we remove this check?

File: public/plugin.tsx [hunk 3]
```diff
@@ -108,7 +116,8 @@ export class AssistantPlugin
           console.warn(`Action executor type ${actionType} is already registered.`);
         actionExecutors[actionType] = execute;
       },
-      assistantEnabled,
+      chatEnabled: () => this.config.chat.enabled,
+      userHasAccess: async () => await getAccount().then(checkAccess),
```
Comment (wanglam) [inline public/plugin.tsx:120]: The result of `userHasAcess` may not the same as the `HeaderChatButton` one. Shall we use the same getAccount result?

File: public/plugin.tsx (hunk 1 above)
Comment (joshuali925) [reply public/plugin.tsx:64]: it's a closure to save http calls, i think it's acceptable that if user permission changes, UI needs a refresh to see the updated results. the real permission check happens on backend anyways

File: public/plugin.tsx (hunk 2 above)
Comment (joshuali925) [reply public/plugin.tsx:63]: not on the second `getAccount()` call

File: public/plugin.tsx (hunk 3 above)
Comment (joshuali925) [reply public/plugin.tsx:120]: they should be the same, account is cached

**Review by wanglam — APPROVED** [review]

**Review by raintygao — APPROVED** [review]

**Pull request merged**